pkg/udev/common.go

```diff
@@ -139,19 +176,21 @@ func (device *UdevDevice) getSize() (uint64, error) {
 // GetDevLinks returns syspath of a disk using syspath we can fell details
 // in diskInfo struct using udev probe
 func (device *UdevDevice) GetDevLinks() map[string][]string {
-	devLinkMap := make(map[string][]string)
-	byIdLink := make([]string, 0)
-	byPathLink := make([]string, 0)
+	devLinkUtil := newDevlinkUtil()
 	for _, link := range strings.Split(device.GetPropertyValue(UDEV_DEVLINKS), " ") {
 		parts := strings.Split(link, "/")
 		if util.Contains(parts, BY_ID_LINK) {
-			byIdLink = append(byIdLink, link)
+			if strings.HasPrefix(parts[4], device.GetPropertyValue(UDEV_BUS)) && strings.HasSuffix(parts[4], device.GetPropertyValue(UDEV_SERIAL_FULL)) {
```

Comment:
Use a variable instead of 4 akin LINK_ID_INDEX.
Also add a comment showing the expected output of the command, so that the index (4) will be self-explanatory.